server/src/main/java/org/spine3/server/entity/FieldMasks.java

```diff
@@ -0,0 +1,172 @@
+        final ProtocolStringList filter = mask.getPathsList();
+
+        final Class<B> builderClass = getBuilderForType(typeUrl);
+
+        if (filter.isEmpty() || builderClass == null) {
+            return Collections.unmodifiableCollection(entities);
+        }
+
+        try {
+            final Constructor<B> builderConstructor = builderClass.getDeclaredConstructor();
+            builderConstructor.setAccessible(true);
+
+            for (Message wholeMessage : entities) {
+                final B builder = builderConstructor.newInstance();
+
+                for (Descriptors.FieldDescriptor field : wholeMessage.getDescriptorForType().getFields()) {
+                    if (filter.contains(field.getFullName())) {
+                        builder.setField(field, wholeMessage.getField(field));
+                    }
+                }
+
+                filtered.add((M) builder.build());
+            }
+
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException e) {
+            // If any reflection failure happens, return all the data without any mask applied.
+            return Collections.unmodifiableCollection(entities);
+        }
+
+        return Collections.unmodifiableList(filtered);
+    }
+
+    /**
+     * <p>Applies the given {@code FieldMask} to a single {@link Message}.
+     **
+     * <p>In case the {@code FieldMask} instance contains invalid field declarations, they are ignored and
+     * do not affect the execution result.
+     *
+     * @param mask {@code FieldMask} instance to apply.
+     * @param entity The {@link Message} to apply given {@code FieldMask} to.
+     * @param typeUrl Type of the {@link Message}.
+     * @return A {@link Message} of the same type as the given one with only selected fields.
+     */
+    @SuppressWarnings("unchecked")
+    public static <M extends  Message, B extends Message.Builder> M applyMask(@SuppressWarnings("TypeMayBeWeakened") FieldMask mask, M entity, TypeUrl typeUrl) {
+        final ProtocolStringList filter = mask.getPathsList();
+
+        final Class<B> builderClass = getBuilderForType(typeUrl);
+
+        if (filter.isEmpty() || builderClass == null) {
+            return entity;
+        }
+
+        try {
+            final Constructor<B> builderConstructor = builderClass.getDeclaredConstructor();
+            builderConstructor.setAccessible(true);
+
+            final B builder = builderConstructor.newInstance();
+
+            for (Descriptors.FieldDescriptor field : entity.getDescriptorForType().getFields()) {
+                if (filter.contains(field.getFullName())) {
+                    builder.setField(field, entity.getField(field));
+                }
+            }
+
+            return (M) builder.build();
+
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException e) {
+            return entity;
+        }
+
+
+    }
```

Comment:
All reasonable scenarios are covered.
We are aiming to cover some of potential exceptional cases as well.